src/services/helpers.ts

```diff
@@ -56,3 +58,31 @@ export const shuffleArray = <T>(array: T[]): T[] => {
   }
   return shuffled
 }
+
+export const getNextExercise = async (
+  profession: Discipline
+): Promise<{ disciplineId: number; exerciseKey: number }> => {
+  const disciplines = await loadDisciplines(profession) // TODO LUN-316 leaf disciplines must be loaded, also if nested
+  if (disciplines.length <= 0) {
+    throw new Error(`No Disciplines for id ${profession.id}`)
+  }
+  const progress = await AsyncStorage.getExerciseProgress()
+  const doneExercisesOfLeafDiscipline = (disciplineId: number): number => {
+    const progressOfDiscipline = progress[disciplineId]
+    return progressOfDiscipline
+      ? Object.keys(progressOfDiscipline).filter(item => progressOfDiscipline[item] !== undefined).length
+      : 0
```

Comment:
Isnt this just the same as `Object.keys.length` or `Object.values.length`?

Reply:
No, as the data structure allows a object like this:

```
200: {
   1: 0.2,
   2: undefined
}

And for this we need length = 1, therefore we need to filter the undefined values.